fuse_constraints/src/absolute_pose_2d_stamped_constraint.cpp

```diff
@@ -37,19 +37,52 @@
 #include <ceres/autodiff_cost_function.h>
 #include <Eigen/Dense>
 
+#include <vector>
+
 
 namespace fuse_constraints
 {
 
 AbsolutePose2DStampedConstraint::AbsolutePose2DStampedConstraint(
   const fuse_variables::Position2DStamped& position,
   const fuse_variables::Orientation2DStamped& orientation,
-  const fuse_core::Vector3d& mean,
-  const fuse_core::Matrix3d& covariance) :
-    fuse_core::Constraint{position.uuid(), orientation.uuid()},
-    mean_(mean),
-    sqrt_information_(covariance.inverse().llt().matrixU())
+  const fuse_core::VectorXd& partial_mean,
+  const fuse_core::MatrixXd& partial_covariance,
+  const std::vector<size_t>& linear_indices,
+  const std::vector<size_t>& angular_indices) :
+    fuse_core::Constraint{position.uuid(), orientation.uuid()}
 {
+  size_t total_variable_size = position.size() + orientation.size();
+  total_indices_ = linear_indices.size() + angular_indices.size();
+
+  assert(partial_mean.rows() == static_cast<int>(total_indices_));
+  assert(partial_covariance.rows() == static_cast<int>(total_indices_));
+  assert(partial_covariance.cols() == static_cast<int>(total_indices_));
+
+  // Compute the sqrt information of the provided cov matrix
+  fuse_core::MatrixXd partial_sqrt_information = partial_covariance.inverse().llt().matrixU();
+
+  // Assemble a mean vector and sqrt information matrix from the provided values, but in proper Variable order
+  // What are we doing here?
+  // The constraint equation is defined as: cost(x) = ||A * (x - b)||^2
+  // If we are measuring a subset of dimensions, we only want to produce costs for the measured dimensions.
+  // But the variable vectors will be full sized. We can make this all work out by creating a non-square A
+  // matrix, where each row computes a cost for one measured dimensions, and the columns are in the order
+  // defined by the variable.
+  mean_ = fuse_core::VectorXd::Zero(total_variable_size);
+  sqrt_information_ = fuse_core::MatrixXd::Zero(total_indices_, total_variable_size);
+  for (size_t i = 0; i < linear_indices.size(); ++i)
+  {
+    mean_(linear_indices[i]) = partial_mean(i);
+    sqrt_information_.col(linear_indices[i]) = partial_sqrt_information.col(i);
+  }
+
+  for (size_t i = linear_indices.size(); i < total_indices_; ++i)
```

Comment:
I recognize that this doesn't need to be in a loop, but it creates parallel structure to the first loop. Happy to change, though.